lib/parser.js

```diff
@@ -14,40 +20,34 @@ const PARCEL_TYPES = {
   PARCEL_STREAM: 5
 };
 
-const constObj = Object.assign({
-  PROTOCOL_VERSION,
-  HANDSHAKE_LENGTH,
-  PARCEL_LENGTH,
-  CHUNK_LENGTH,
-  STRUCT_PARCEL: 0,
-  STRUCT_CHUNK: 1,
-}, PARCEL_TYPES);
-
-const readParcel = buffer => {
+const readParcel = (
   // Read parcel from buffer
+  buffer // [buffer type]
   // Return parcel object
+) => {
   const rest = Buffer.from(buffer, 0, PARCEL_LENGTH);
-  const parcel = { structType: constObj.STRUCT_PARCEL };
-  parcel.parcelId = rest.readInt32LE(1);
-  parcel.parcelType = rest.readInt8(5);
-  parcel.compression = rest.readInt8(6);
-  parcel.encoding = rest.readInt8(7);
-  const parcelLength = rest.slice(8).toString();
-  parcel.length = parseInt(parcelLength);
+  const parcel = {
+    structType: constants.STRUCT_PARCEL,
+    parcelId: rest.readInt32LE(1),
+    parcelType: rest.readInt8(5),
+    compression: rest.readInt8(6),
+    encoding: rest.readInt8(7),
+    length: parseInt(rest.slice(8).toString()),
+  };
   return parcel;
 };
 
 const parcel = ({
-  // Encode parcel structure to Buffer
-  parcelId,           // id, // parcelId: 4b
-  parcelType,         // type, // parcelType: 1b
+// Encode parcel structure to Buffer
+  parcelId,     // id, // parcelId: 4b
```

Comment:
Our functions comments style looks like this:
```js
parameterName // type[or otherType], short description IF needed
```
e.g. assuming that `parcelId` type is number, you should have something like this:
```js
parcelId, // number, id of parcel
```
If you want, you can add info about its size

Reply:
BTW, why are taking parameters using destructuring assignment?

Reply:
With this syntax we can describe all parameters that we use in parsing.

Reply:
As for me, something like this will be more appropriate
```js
const parcel = (
   parcelId,
   parcelType,
   compression,
   encoding,
   length
) => {
```
Or there is some libraries, for which we need to save backwards compatibility? If that is the case. let me know

Reply:
IMHO, destructuring here is more comfortable as we form parcel object, taking into consideration struct scheme, and then pass it.
So, without destructuring instead of this in connection:
```javascript
const parcel = {
  parcelId: <parcelId>,
  parcelType: <parcelType>
  compression: <compression>,
  length: <length>
};
parser.parcel(parcel);
```
we will have to write smth like this:
```javascript
parser.parcel(parcel.parcelId, parcel.parcelType...);
```
if i got your point correctly.
Or don`t use options as object and declare all options separately, which, FWIW, is less descriptive

Reply:
I agree with having one object as parameter, though in this case I would rather have this
```js
const parcel = (
   // Encode parcel structure to Buffer
   param // object { parcelId, parcelType, compression, encoding, length}
   //   parcelId - number, id of parcel
   //   parcelType - number, type of parcel
   //   compression - number, compression type
   //   encoding - number, encoding type
   //   length - string
   // Returns: buffer
) => {
```